File: js/math/gcd-lcm.js
function gcd(a,b){
	var m;
	while(b!=0){
		m=a%b;
		a=b;
		b=m;
	}
	return a;
}

function lcm(a,b){
	var m = (a*b)/gcd(a,b);
	return m;
}

console.log(gcd(664,414));
console.log(lcm(664,414));

File: java/math/GcdLcm.java
class GcdLcm
{
    static int gcd(int a,int b){
    	int m;
    	while(b!=0){
    		m=a%b;
    		a=b;
    		b=m;
    	}
    	return a;
    }
    
    static int lcm(int a,int b){
    	int m = (a*b)/gcd(a,b);
    	return m;
    }

	public static void main (String[] args) throws java.lang.Exception
	{
		System.out.println(gcd(664,414));
		System.out.println(lcm(664,414));
	}
}